python/ray/_private/usage/usage_lib.py

```diff
@@ -135,6 +140,50 @@ class UsageStatsEnabledness(Enum):
     ENABLED_BY_DEFAULT = auto()
 
 
+_recorded_library_usages = set()
+
+
+def _put_library_usage(library_usage: str):
+    assert _internal_kv_initialized()
+    try:
+        ray._private.utils.internal_kv_put_with_retry(
```

Comment:
Why do we need to retry? That might block; prefer to fast fail as this path isn't critical.